src/components/WorkBuddies/WorkBuddies.tsx

```diff
@@ -85,13 +90,15 @@ const WorkBuddies = () => {
       return;
     }
 
-    const updatedBuddies = buddies;
-    updatedBuddies.forEach((buddy) => {
+    const updatedBuddies = buddies.map((buddy) => {
       if (buddy.url === myBuddy.url) {
-        if (nameRef.current?.value) {
-          buddy.name = nameRef.current!.value;
-        }
-        buddy.isMine = true;
+        return {
+          ...buddy,
+          name: nameRef.current?.value ? nameRef.current?.value : buddy.name,
```

Comment:
I'm expecting here the `name` variable to be used instead of the nameRef.current?.value